x-pack/plugins/security_solution/public/common/components/event_details/table/use_action_cell_data_provider.ts

```diff
@@ -93,6 +95,23 @@ export const useActionCellDataProvider = ({
   const cellData = useMemo(() => {
     if (values === null || values === undefined) return null;
     const arrayValues = Array.isArray(values) ? values : [values];
+
+    // For fields with multiple values we need add an extra filter that makes sure
+    // that only fields that match ALL the values are queried later on.
+    let filters: Filter[] = [];
+    if (arrayValues.length > 1) {
+      filters = [
+        {
+          meta: {},
+          query: {
+            bool: {
+              must: arrayValues.map((value) => ({ term: { [field]: value } })),
+            },
+          },
+        },
+      ];
+    }
```

Comment:
Not sure this logic should apply in cases other than alert prevalence. But I might just leave this here instead of in `PrevalenceCell` so that in future this logic can be used in other places.